bpf/custom/README.rst

```diff
@@ -0,0 +1,146 @@
+  packet, and must preserve the return value selected by the datapath.
+
+- The identity corresponding to the packet, that is, the Cilium identity
+  associated to the sender for the ingress hook, and to the receiver when on
+  the egress hook. This identity is not strictly necessary to the custom
+  program, but it prove useful for some use cases. Given that the datapath
+  already did the work for retrieving the identity for the packet, it just
+  passes it down to the custom programs, in case it saves them the trouble to
+  extract it a second time.
+
+Limitations
+-----------
+
+- There is currently no hook on the egress path for socket-based
+  load-balancing.
+
+Sample Byte Counter Program
+============================
+
+This directory contains a sample byte counter program, to help count how many
+bytes a given endpoint sends or receives, and to understand how the tail call
+hooks work.
+
+The following steps indicate how to use this example program.
+
+1. Compile the program. This should be easy thanks to the provided Makefile.
+   From the root of the Cilium repository::
+
+       $ make -C bpf/custom
+
+2. Load and pin the byte counter program. Bpftool can do it::
+
+       # bpftool prog load bpf/custom/bpf_custom.o \
+           /sys/fs/bpf/tc/globals/bytecounter \
+           type classifier \
+           pinmaps /sys/fs/bpf/tc/globals/bytecounter_maps
+
+3. Pick the endpoint for which the program should count the bytes. Update the
+   tail call map for custom programs for that endpoint, with a reference to the
+   pinned byte counter program::
+
+       $ EP=<endpoint_id>
+       # bpftool map update \
+           pinned /sys/fs/bpf/tc/globals/cilium_calls_custom_$(printf '%05d' ${EP}) \
+           key 0 0 0 0 \
+           value pinned /sys/fs/bpf/tc/globals/bytecounter
+
+   The key indicate the hook to use. It is possible to attach the program at
+   multiple hooks, although the counter currently makes no distinction and will
+   condensate the values collected from all hooks into a single counter. The
+   available hooks and their keys are the following:
+
+   - IPv4, ingress: ``key 0 0 0 0``
+   - IPv4, egress:  ``key 1 0 0 0``
+   - IPv6, ingress: ``key 2 0 0 0``
+   - IPv6, egress:  ``key 3 0 0 0``
+
+4. After some traffic has flown, dump the content of the byte counter map to
+   read the statistics::
+
+       # bpftool map dump \
+           pinned /sys/fs/bpf/tc/globals/bytecounter_maps/bytecount_map
+
+5. When the byte counter is no longer neccessary, it is possible to clean it up
+   by deleting the entry in the tail call map, and removing the pinned
+   objects::
+
+       # bpftool map delete \
+           pinned /sys/fs/bpf/tc/globals/cilium_calls_custom_$(printf '%05d' ${EP}) \
+           key 0 0 0 0
+       # rm /sys/fs/bpf/tc/globals/custom_prog
+       # rm -r /sys/fs/bpf/tc/globals/bytecounter_maps
+
+Using Other Custom Programs
+===========================
```

Comment:
nit: Are these instructions substantially different than the above? At a glance they seem very similar and I figure anyone writing additional custom statistics programs would likely just modify this custom example to get where they're going.

Not a big deal, I'd personally probably just skip the instructions below but if you think they should stay then feel free to just mark this comment as resolved.

Reply:
Makes sense. I kept the section but simplified it and removed all the duplicated instructions (essentially keeping the `make` command with the environment variables to set a different name for the file to include).